.github/workflows/publish-vapor-image.yml

```diff
@@ -0,0 +1,63 @@
+name: Publish Vapor image to ECR
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  publish:
+    name: Publish Vapor Image
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8
+
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@57396166ad8aefe6098280995947635806a0e6ea
+        with:
+          images: |
+            name=bookstore-vapor
+          tags: |
+            # minimal (short sha)
+            type=sha
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1-node16
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: eu-west-1
+      
+      - name: Login to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@261a7de32bda11ba01f4d75c4ed6caf3739e54be
+      
+      - name: Resolve Dependencies
+        run: swift package resolve
+        working-directory: ./bookstore-vapor
+
+      - name: Build
+        run: swift build -c release -Xswiftc -O --static-swift-stdlib
```

Comment:
I would suggest creating a Multi Stage Dockerfile. In the first stage we could build the swift app and in the second have the code to run it.

Reply:
@akoufa: could you tell me what the advantage is for using a multi stage docker file.
It is the way Vapors template works, but I found a few more advantages in doing it manually( eg. private dependencies and caching).

Am just wondering

Reply:
@elmoritz Yes you will get a smaller image in the end. Because the first part of the image the builder will have all the dependencies and tools needed to build your code and the second part of the image will have the application to run with only the dependencie needed for running it. Also you get improvements in caching the different docker layers of the image resulting in faster builds and the everything needed to build and run will be encoded in a single Dockerfile

Reply:
DONE